OpenSim/Simulation/MarkersReference.h

```diff
@@ -105,13 +105,17 @@ class OSIMSIMULATION_API MarkersReference : public Reference_<SimTK::Vec3> {
     //--------------------------------------------------------------------------
     MarkersReference();
 
-    /** Convenience load markers from a file */
+    /** Convenience load markers from a file. See below. */
     MarkersReference(const std::string& markerFileName,
+                     const Set<MarkerWeight>* markerWeightSet = nullptr,
                      Units modelUnits = Units(Units::Meters));
```

Comment:
I understand using a pointer is to allow a default value for nullptr, but is that an actual use case in our code base? If not I would suggest passing a const reference to `Set<MarkerWeight>` instead

Reply:
Yes the InverseKinematicsStudy (does not use IKTasks) uses the default arguments and only provides the filename to the constructor.
Also, the interface was minimally changed and made to be consistent with the other convenience constructor that takes a `TimeSeriesTable` (instead of file name) and already used a pointer to the `MarkerWeightSet`. I agree a const reference is a bit cleaner, but then it requires the caller to create an empty one.  What's your preference?

Reply:
Considering where we are, I'd leave it as is to minimize change especially if not used in InverseKinematicsStudy.  It just stood out looking at the MarkerReference implementation the mixed use of references and pointers which is the price we pay.

Reply:
I went ahead and changed it for consistency. All arguments are const references.